src/Security/PasswordResetService.php

```diff
@@ -17,7 +17,9 @@ public function __construct(
     public function isPasswordRequestTokenValid(LocalAccount $auth, string $token): bool
     {
         $encoder = $this->passwordHasher->getPasswordHasher($auth);
-        $valid = $encoder->verify($auth->getPasswordRequestToken(), $token);
+        $passwordToken = $auth->getPasswordRequestToken();
+        assert($passwordToken !== null);
+        $valid = $encoder->verify($passwordToken, $token);
```

Comment:
How about the shorter and equally functional:
```php
$valid = $encoder->verify($auth->getPasswordRequestToken() ?? '', $token);
```
This removes the need for an assert.